packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -131,21 +138,22 @@ export const CalendarMonth = ({
   const isValidated = (date: Date) => validators.every(validator => validator(date));
   const focusedDateValidated = isValidated(focusedDate);
   useEffect(() => {
-    if (!(isValidDate(dateProp) && isSameDate(focusedDate, dateProp))) {
+    if (isValidDate(dateProp) && isSameDate(focusedDate, dateProp)) {
       setFocusedDate(dateProp);
```

Comment:
As-is, the calendar month page jumps to that last example since focus() is being called, and when clicking a date on the Selectable example and then trying to use the arrow keys focus isn't placed on the correct date.

Having `setFocusedDate(dateProp`only get called if the dateProp is valid and if `isSameDate()` returns false should resolve both those issues.